src/P2P/TaskQueue.hs

```diff
@@ -140,12 +143,22 @@ instance Ord (Task env a) where
 -- enough to be useful in other contexts, too.
 --
 session
+    :: AttemptsCount
+    -> PQueue (Task env a)
+    -> LogFunction
+    -> env
+    -> PeerInfo
+    -> IO Bool
+session li q lo e _ = session_ li q lo e
```

Comment:
Why not consolidate on a single `session` function, if it's just redirecting like this anyway?

Reply:
Ah, it's that the `P2pSession` type changed. Hm.

Reply:
The `session_` function is there mostly to limit the amount of code changes and not have to wrap `PeerInfo` in a `Maybe` for situations when it's not available.